--- busybox-1.22.1.orig/coreutils/sleep.c	2014-01-09 19:15:44.000000000 +0100
+++ busybox-1.22.1/coreutils/sleep.c	2014-03-20 11:31:27.568935920 +0100
@@ -37,6 +37,8 @@
 
 /* Do not make this applet NOFORK. It breaks ^C-ing of pauses in shells */
 
+/* Busybox config mechanism seems to be broken for FLOAT_SLEEP option*/
+#define ENABLE_FEATURE_FLOAT_SLEEP 1
 
 #if ENABLE_FEATURE_FANCY_SLEEP || ENABLE_FEATURE_FLOAT_SLEEP
 static const struct suffix_mult sfx[] = {
