--- system.dts.old	2015-07-23 23:14:55.434444209 +0200
+++ system.dts	2015-07-23 23:05:18.198460862 +0200
@@ -28,6 +28,22 @@
 		device_type = "memory";
 		reg = <0x0 0x20000000>;
 	};
+	reserved-memory {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		ranges;
+		linux,cma {
+			compatible = "shared-dma-pool";
+			reusable;
+			size = <0x1000000>;
+			alignment = <0x2000>;
+			linux,cma-default;
+		};
+		rprx_reserverd: labuf@1c000000 {
+			/*compatible = "redpitaya,rprx-memory";*/
+			reg = <0x1c000000 0x2000000>;
+		};
+	};
 };
 &gem0 {
 	local-mac-address = [00 0a 35 00 00 00];
@@ -86,3 +102,79 @@
 	fclk-enable = <0xf>;
 	ps-clk-frequency = <33333333>;
 };
+&gem0 {
+	phy-handle = <&ethernet_0_mdio>;
+	ethernet_0_mdio: mdio@1 {
+		compatible = "lantiq,phy11g";
+		device_type = "ethernet-phy";
+		reg = <1>;
+	};
+};
+&i2c0 {
+	eep@50 {
+		compatible = "24c64";
+		reg = <0x50>;
+	};
+};
+&spi1 {
+	spidev@0 {
+		compatible = "spidev";
+		reg = <0>; //chipselect 0
+		spi-max-frequency = <50000000>;
+	};
+};
+&clkc {
+	fclk-enable = <0xf>;
+};
+&xadc_wiz_0 {
+	compatible = "xlnx,axi-xadc-1.00.a";
+	clocks = <&clkc 18>;
+	xlnx,channels {
+		#address-cells = <1>;
+		#size-cells = <0>;
+		channel@0 {
+			reg = <0>;
+		};
+		channel@1 {
+			reg = <1>;
+		};
+		channel@2 {
+			reg = <2>;
+		};
+		channel@9 {
+			reg = <9>;
+		};
+		channel@10 {
+			reg = <10>;
+		};
+	};
+};
+/ {
+	gpio-leds {
+		compatible = "gpio-leds";
+		led-8-yellow {
+			label = "led8";
+			gpios = <&gpio0 0 0>;
+			default-state = "off";
+			linux,default-trigger = "mmc0";
+		};
+		led-9-red {
+			label = "led9";
+			gpios = <&gpio0 7 0>;
+			default-state = "off";
+			linux,default-trigger = "heartbeat";
+		};
+	};
+};
+/ {
+	usb_phy0: phy0 {
+		compatible = "ulpi-phy";
+		#phy-cells = <0>;
+		reg = <0xe0002000 0x1000>;
+		view-port = <0x0170>;
+		drv-vbus;
+	};
+};
+&usb0 {
+	usb-phy = <&usb_phy0>;
+};
