--- system.dts.old	2016-05-06 01:01:41.187394113 +0200
+++ system.dts	2016-05-06 01:14:43.282854613 +0200
@@ -26,7 +26,7 @@
 	};
 	memory {
 		device_type = "memory";
-		reg = <0x0 0x20000000>;
+		reg = <0x0 0x1c000000>;
 	};
 };
 &gem0 {
@@ -86,3 +86,89 @@
 	fclk-enable = <0xf>;
 	ps-clk-frequency = <33333333>;
 };
+
+&gem0 {
+	phy-handle = <&ethernet_0_mdio>;
+	ethernet_0_mdio: mdio@1 {
+		compatible = "lantiq,phy11g";
+		device_type = "ethernet-phy";
+		reg = <1>;
+	};
+};
+&i2c0 {
+	eep@50 {
+		compatible = "24c64";
+		reg = <0x50>;
+	};
+};
+&spi1 {
+	spidev@0 {
+		compatible = "spidev";
+		reg = <0>; //chipselect 0
+		spi-max-frequency = <50000000>;
+	};
+};
+&xadc {
+	compatible = "xlnx,axi-xadc-1.00.a";
+	clocks = <&clkc 18>;
+	xlnx,channels {
+		#address-cells = <1>;
+		#size-cells = <0>;
+		channel@0 {
+			reg = <0>;
+		};
+		channel@1 {
+			reg = <1>;
+		};
+		channel@2 {
+			reg = <2>;
+		};
+		channel@9 {
+			reg = <9>;
+		};
+		channel@10 {
+			reg = <10>;
+		};
+	};
+};
+/ {
+	gpio-leds {
+		compatible = "gpio-leds";
+		led-8-yellow {
+			label = "led8";
+			gpios = <&gpio0 0 0>;
+			default-state = "off";
+			linux,default-trigger = "mmc0";
+		};
+		led-9-red {
+			label = "led9";
+			gpios = <&gpio0 7 0>;
+			default-state = "off";
+			linux,default-trigger = "heartbeat";
+		};
+	};
+};
+/ {
+	usb_0_phy: phy@e0002000 {
+		compatible = "ulpi-phy";
+		#phy-cells = <0>;
+		reg = <0xe0002000 0x1000>;
+		view-port = <0x0170>;
+		drv-vbus;
+	};
+};
+&usb0 {
+	usb-phy = <&usb_0_phy>;
+};
+
+/ {
+	#address-cells = <1>;
+	#size-cells = <1>;
+	#interrupt-cells = <3>;
+	snd_rp_ac97: redpitaya-ac97@40700000 {
+		compatible = "redpitaya,redpitaya-ac97";
+		reg = <0x40700000 0x100>;
+		interrupts = <0 30 1>, <0 31 1>;
+		interrupt-parent = <&intc>;
+	};
+};
