--- system.dts.old	2015-07-23 23:14:55.434444209 +0200
+++ system.dts	2015-07-23 23:05:18.198460862 +0200
@@ -26,7 +26,7 @@
 	};
 	memory {
 		device_type = "memory";
-		reg = <0x0 0x20000000>;
+		reg = <0x0 0x1c000000>;
 	};
 };
 &gem0 {
@@ -86,3 +86,144 @@
 	fclk-enable = <0x0>;
 	ps-clk-frequency = <33333333>;
 };
+
+&gem0 {
+	phy-handle = <&phy0>;
+	ps7_ethernet_0_mdio: mdio {
+		phy0: phy@1 {
+			compatible = "lantiq,phy11g";
+			device_type = "ethernet-phy";
+			reg = <32>;
+		};
+	};
+};
+&i2c0 {
+	eep@50 {
+		compatible = "24c64";
+		reg = <0x50>;
+	};
+};
+&spi1 {
+	spidev@0 {
+		compatible="spidev";
+		reg =<0>; //chipselect 0
+		spi-max-frequency= <50000000>;
+	};
+};
+&clkc {
+	fclk-enable = <0xf>;
+};
+/* &xadc {
+	compatible = "xlnx,axi-xadc-1.00.a";
+	clocks = <&clkc 18>;
+	xlnx,channels {
+		#address-cells = <1>;
+		#size-cells = <0>;
+		channel@0 {
+			reg = <0>;
+		};
+		channel@1 {
+			reg = <1>;
+		};
+		channel@2 {
+			reg = <2>;
+		};
+		channel@9 {
+			reg = <9>;
+		};
+		channel@10 {
+			reg = <10>;
+		};
+	};
+}; */
+/ {
+	gpio-leds {
+	    compatible = "gpio-leds";
+	    led-8-yellow {
+	        label = "led8";
+	        gpios = <&gpio0 0 0>;
+	        default-state = "off";
+		linux,default-trigger = "mmc0";
+	    };
+	    led-9-red {
+	        label = "led9";
+	        gpios = <&gpio0 7 0>;
+	        default-state = "off";
+		linux,default-trigger = "hartbeat";
+	    };
+	};
+};
+/ {
+	usb_phy0: phy0 {
+		compatible = "ulpi-phy";
+		#phy-cells = <0>;
+		reg = <0xe0002000 0x1000>;
+		view-port = <0x0170>;
+		drv-vbus;
+	};
+};
+&usb0 {
+	usb-phy = <&usb_phy0>;
+};
+/* / {
+	dmatest_0: dmatest@0 {
+		compatible ="xlnx,axi-dma-test-1.00.a";
+		dmas = <&axi_dma_3 0
+			&axi_dma_3 1>;
+		dma-names = "axidma0", "axidma1";
+	};
+}; */
+/ {
+	housekeeping@40000000 {
+		compatible = "generic-uio";
+		reg = <0x40000000 0x10000>;
+	};
+	muxctl@40040000 {
+		compatible = "generic-uio";
+		reg = <0x40040000 0x10000>;
+	};
+	gpio@40080000 {
+		compatible = "generic-uio";
+		reg = <0x40080000 0x10000>;
+	};
+	led@400c0000 {
+		compatible = "generic-uio";
+		reg = <0x400c0000 0x10000>;
+	};
+	calib@40100000 {
+		compatible = "generic-uio";
+		reg = <0x40100000 0x10000>;
+	};
+	pdm@40140000 {
+		compatible = "generic-uio";
+		reg = <0x40140000 0x10000>;
+	};
+	pwm@40180000 {
+		compatible = "generic-uio";
+		reg = <0x40180000 0x10000>;
+	};
+	asg0@401c0000 {
+		compatible = "generic-uio";
+		reg = <0x401c0000 0x10000>;
+	};
+	asg1@40200000 {
+		compatible = "generic-uio";
+		reg = <0x40200000 0x10000>;
+	};
+	scope0@40240000 {
+		compatible = "generic-uio";
+		reg = <0x40240000 0x10000>;
+	};
+	scope1@40280000 {
+		compatible = "generic-uio";
+		reg = <0x40280000 0x10000>;
+	};
+	lg@402c0000 {
+		compatible = "generic-uio";
+		reg = <0x402c0000 0x10000>;
+	};
+	la@40300000 {
+		compatible = "generic-uio";
+		reg = <0x40300000 0x10000>;
+	};
+};
