--- system.dts.old	2015-07-23 23:14:55.434444209 +0200
+++ system.dts	2015-07-23 23:05:18.198460862 +0200
@@ -28,6 +28,22 @@
 		device_type = "memory";
 		reg = <0x0 0x20000000>;
 	};
+	reserved-memory {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		ranges;
+		linux,cma {
+			compatible = "shared-dma-pool";
+			reusable;
+			size = <0x1000000>;
+			alignment = <0x2000>;
+			linux,cma-default;
+		};
+		rprx_reserverd: labuf@1c000000 {
+			/*compatible = "redpitaya,rprx-memory";*/
+			reg = <0x1c000000 0x2000000>;
+		};
+	};
 };
 &gem0 {
 	local-mac-address = [00 0a 35 00 00 00];
@@ -86,3 +102,185 @@
 	fclk-enable = <0xf>;
 	ps-clk-frequency = <33333333>;
 };
+&gem0 {
+	phy-handle = <&ethernet_0_mdio>;
+	ethernet_0_mdio: mdio@1 {
+		compatible = "lantiq,phy11g";
+		device_type = "ethernet-phy";
+		reg = <1>;
+	};
+};
+&i2c0 {
+	eep@50 {
+		compatible = "24c64";
+		reg = <0x50>;
+	};
+gpio@20 {
+	compatible = "nxp,pca9557";
+	reg = <0x1a>;
+/*	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_pca9505>;
+	interrupt-parent = <&gpio3>;
+	interrupts = <23 IRQ_TYPE_LEVEL_LOW>;*/
+};
+
+i2c-mux@75 {
+	compatible = "nxp,pca9547";
+	#address-cells = <1>;
+	#size-cells = <0>;
+	reg = <0x70>;
+	status="okay";	
+	i2c@1 {
+		#address-cells = <1>;
+		#size-cells = <0>;
+		reg = <1>;
+		gpio@20 {
+			ompatible = "nxp,pca9557";
+			reg = <0x1a>;
+		};
+		pca9555_20: pca9555@20 {
+			compatible = "nxp,pca9555";
+			gpio-controller;
+			#gpio-cells = <2>;
+			reg = <0x20>;
+		};
+		pca9555_21: pca9555@21 {
+			compatible = "nxp,pca9555";
+			gpio-controller;
+			#gpio-cells = <2>;
+			reg = <0x21>;
+		};
+};
+   i2c@2 {
+            #address-cells = <1>;
+             #size-cells = <0>;
+              reg = <2>;
+               gpio@20 {
+			compatible = "nxp,pca9557";
+			reg = <0x1a>;
+	};
+  };
+   i2c@3 {
+            #address-cells = <1>;
+             #size-cells = <0>;
+              reg = <3>;
+               gpio@20 {
+			compatible = "nxp,pca9557";
+			reg = <0x1a>;
+	};
+  };
+   i2c@4 {
+            #address-cells = <1>;
+             #size-cells = <0>;
+              reg = <4>;
+               gpio@20 {
+			compatible = "nxp,pca9557";
+			reg = <0x1a>;
+		};
+       };
+        i2c@5 {
+       #address-cells = <1>;
+        #size-cells = <0>;
+         reg = <5>;
+  };
+   i2c@6 {
+            #address-cells = <1>;
+             #size-cells = <0>;
+              reg = <6>;
+               gpio@20 {
+			compatible = "nxp,pca9557";
+			reg = <0x1a>;
+		};
+       };
+        i2c@7 {
+    #address-cells = <1>;
+     #size-cells = <0>;
+      reg = <7>;
+       gpio@20 {
+		compatible = "nxp,pca9557";
+		reg = <0x1a>;
+	};
+  };
+   i2c@8 {
+            #address-cells = <1>;
+             #size-cells = <0>;
+              reg = <8>;
+               gpio@20 {
+			compatible = "nxp,pca9557";
+			reg = <0x1a>;
+		};
+  };
+   i2c@9 {
+            #address-cells = <1>;
+             #size-cells = <0>;
+              reg = <9>;
+               gpio@20 {
+			compatible = "nxp,pca9557";
+			reg = <0x1a>;
+		};
+       };
+
+};
+};
+&spi1 {
+	spidev@0 {
+		compatible = "spidev";
+		reg = <0>; //chipselect 0
+		spi-max-frequency = <50000000>;
+	};
+};
+&clkc {
+	fclk-enable = <0xf>;
+};
+&xadc_wiz_0 {
+	compatible = "xlnx,axi-xadc-1.00.a";
+	clocks = <&clkc 18>;
+	xlnx,channels {
+		#address-cells = <1>;
+		#size-cells = <0>;
+		channel@0 {
+			reg = <0>;
+		};
+		channel@1 {
+			reg = <1>;
+		};
+		channel@2 {
+			reg = <2>;
+		};
+		channel@9 {
+			reg = <9>;
+		};
+		channel@10 {
+			reg = <10>;
+		};
+	};
+};
+/ {
+	gpio-leds {
+		compatible = "gpio-leds";
+		led-8-yellow {
+			label = "led8";
+			gpios = <&gpio0 0 0>;
+			default-state = "off";
+			linux,default-trigger = "mmc0";
+		};
+		led-9-red {
+			label = "led9";
+			gpios = <&gpio0 7 0>;
+			default-state = "off";
+			linux,default-trigger = "heartbeat";
+		};
+	};
+};
+/ {
+	usb_phy0: phy0 {
+		compatible = "ulpi-phy";
+		#phy-cells = <0>;
+		reg = <0xe0002000 0x1000>;
+		view-port = <0x0170>;
+		drv-vbus;
+	};
+};
+&usb0 {
+	usb-phy = <&usb_phy0>;
+};
