diff -rupN old/linux-xlnx-xilinx-v2015.1/arch/arm/configs/xilinx_zynq_defconfig linux-xlnx-xilinx-v2015.1/arch/arm/configs/xilinx_zynq_defconfig
--- old/linux-xlnx-xilinx-v2015.1/arch/arm/configs/xilinx_zynq_defconfig	2015-03-03 09:24:37.000000000 +0100
+++ linux-xlnx-xilinx-v2015.1/arch/arm/configs/xilinx_zynq_defconfig	2015-05-07 12:39:23.701748413 +0200
@@ -117,6 +117,7 @@ CONFIG_XILINX_AXI_EMAC=y
 CONFIG_XILINX_PS_EMAC=y
 CONFIG_MARVELL_PHY=y
 CONFIG_VITESSE_PHY=y
+CONFIG_LANTIQ_PHY=y
 CONFIG_MDIO_BITBANG=y
 CONFIG_INPUT_SPARSEKMAP=y
 CONFIG_INPUT_EVDEV=y
@@ -245,3 +246,39 @@ CONFIG_RCU_CPU_STALL_TIMEOUT=60
 CONFIG_FONTS=y
 CONFIG_FONT_8x8=y
 CONFIG_FONT_8x16=y
+CONFIG_WIRELESS=y
+CONFIG_CFG80211=y
+CONFIG_MAC80211=y
+CONFIG_RTLWIFI=y
+CONFIG_RTL8192CU=y
+CONFIG_RT2X00=y
+CONFIG_RT2800USB=y
+CONFIG_RT2800USB_RT3573=y
+CONFIG_RT2800USB_RT53XX=y
+CONFIG_RT2800USB_RT55XX=y
+CONFIG_RT2800USB_UNKNOWN=y
+CONFIG_INPUT_TOUCHSCREEN=y
+CONFIG_OF_TOUCHSCREEN=y
+CONFIG_TOUCHSCREEN_USB_COMPOSITE=y
+CONFIG_TOUCHSCREEN_USB_EGALAX=y
+CONFIG_TOUCHSCREEN_USB_PANJIT=y
+CONFIG_TOUCHSCREEN_USB_3M=y
+CONFIG_TOUCHSCREEN_USB_ITM=y
+CONFIG_TOUCHSCREEN_USB_ETURBO=y
+CONFIG_TOUCHSCREEN_USB_GUNZE=y
+CONFIG_TOUCHSCREEN_USB_DMC_TSC10=y
+CONFIG_TOUCHSCREEN_USB_IRTOUCH=y
+CONFIG_TOUCHSCREEN_USB_IDEALTEK=y
+CONFIG_TOUCHSCREEN_USB_GENERAL_TOUCH=y
+CONFIG_TOUCHSCREEN_USB_GOTOP=y
+CONFIG_TOUCHSCREEN_USB_JASTEC=y
+CONFIG_TOUCHSCREEN_USB_ELO=y
+CONFIG_TOUCHSCREEN_USB_E2I=y
+CONFIG_TOUCHSCREEN_USB_ZYTRONIC=y
+CONFIG_TOUCHSCREEN_USB_ETT_TC45USB=y
+CONFIG_TOUCHSCREEN_USB_NEXIO=y
+CONFIG_TOUCHSCREEN_USB_EASYTOUCH=y
+CONFIG_FB_SYS_FOPS=y
+CONFIG_FB_DEFERRED_IO=y
+CONFIG_FB_MODE_HELPERS=y
+CONFIG_FB_UDL=y
diff -rupN old/linux-xlnx-xilinx-v2015.1/drivers/misc/eeprom/at24.c linux-xlnx-xilinx-v2015.1/drivers/misc/eeprom/at24.c
--- old/linux-xlnx-xilinx-v2015.1/drivers/misc/eeprom/at24.c	2015-03-03 09:24:37.000000000 +0100
+++ linux-xlnx-xilinx-v2015.1/drivers/misc/eeprom/at24.c	2015-05-07 12:37:40.173235042 +0200
@@ -508,6 +508,9 @@ static int at24_probe(struct i2c_client
 		 * is recommended anyhow.
 		 */
 		chip.page_size = 1;
+		if(!strncmp(id->name,"24c64",5)){
+			chip.page_size = 32;
+		}
 
 		/* update chipdata if OF is present */
 		at24_get_ofdata(client, &chip);
diff -rupN old/linux-xlnx-xilinx-v2015.1/drivers/net/phy/Kconfig linux-xlnx-xilinx-v2015.1/drivers/net/phy/Kconfig
--- old/linux-xlnx-xilinx-v2015.1/drivers/net/phy/Kconfig	2015-03-03 09:24:37.000000000 +0100
+++ linux-xlnx-xilinx-v2015.1/drivers/net/phy/Kconfig	2015-05-07 12:37:40.173235042 +0200
@@ -118,6 +118,11 @@ config MICREL_PHY
 	---help---
 	  Supports the KSZ9021, VSC8201, KS8001 PHYs.
 
+config LANTIQ_PHY
+	tristate "Driver for Lantiq PHYs"
+	---help---
+	  Supports the 11G and 22E PHYs.
+
 config FIXED_PHY
 	bool "Driver for MDIO Bus/PHY emulation with fixed speed/link PHYs"
 	depends on PHYLIB=y
diff -rupN old/linux-xlnx-xilinx-v2015.1/drivers/net/phy/Makefile linux-xlnx-xilinx-v2015.1/drivers/net/phy/Makefile
--- old/linux-xlnx-xilinx-v2015.1/drivers/net/phy/Makefile	2015-03-03 09:24:37.000000000 +0100
+++ linux-xlnx-xilinx-v2015.1/drivers/net/phy/Makefile	2015-05-07 12:37:40.173235042 +0200
@@ -24,6 +24,7 @@ obj-$(CONFIG_NATIONAL_PHY)	+= national.o
 obj-$(CONFIG_DP83640_PHY)	+= dp83640.o
 obj-$(CONFIG_STE10XP)		+= ste10Xp.o
 obj-$(CONFIG_MICREL_PHY)	+= micrel.o
+obj-$(CONFIG_LANTIQ_PHY)	+= lantiq.o
 obj-$(CONFIG_MDIO_OCTEON)	+= mdio-octeon.o
 obj-$(CONFIG_MICREL_KS8995MA)	+= spi_ks8995.o
 obj-$(CONFIG_AT803X_PHY)	+= at803x.o
diff -rupN old/linux-xlnx-xilinx-v2015.1/drivers/uio/uio_pdrv_genirq.c linux-xlnx-xilinx-v2015.1/drivers/uio/uio_pdrv_genirq.c
--- old/linux-xlnx-xilinx-v2015.1/drivers/uio/uio_pdrv_genirq.c	2015-03-03 09:24:37.000000000 +0100
+++ linux-xlnx-xilinx-v2015.1/drivers/uio/uio_pdrv_genirq.c	2015-05-07 12:37:40.173235042 +0200
@@ -148,7 +148,7 @@ static int uio_pdrv_genirq_probe(struct
 	if (!uioinfo->irq) {
 		ret = platform_get_irq(pdev, 0);
 		uioinfo->irq = ret;
-		if (ret == -ENXIO && pdev->dev.of_node)
+		if (ret == -EINVAL && pdev->dev.of_node)
 			uioinfo->irq = UIO_IRQ_NONE;
 		else if (ret < 0) {
 			dev_err(&pdev->dev, "failed to get IRQ\n");
