diff -rupN lua-nginx-module.old/config lua-nginx-module.new/config
--- lua-nginx-module.old/config	2013-09-02 21:58:33.000000000 +0200
+++ lua-nginx-module.new/config	2015-06-12 14:39:57.279348839 +0200
@@ -6,6 +6,15 @@ ngx_feature_incs="#include <lauxlib.h>"
 ngx_feature_path=
 ngx_feature_test="(void) luaL_newstate();"
 
+LUAJIT_INC="-I ${SYSROOT}/usr/include -I ${SYSROOT}/usr/include/luajit-2.0 ";
+LUAJIT_LIB="-L ${SYSROOT}/usr/lib -lluajit-5.1";
+ngx_force_module=yes;
+
+if test -n "$ngx_force_module"; then
+   ngx_found=yes;
+   ngx_feature_path=$LUAJIT_INC;
+   ngx_feature_libs=$LUAJIT_LIB;
+else
 if [ -n "$LUAJIT_INC" -o -n "$LUAJIT_LIB" ]; then
     # explicitly set Lua lib path
     ngx_feature="LuaJIT library in $LUAJIT_LIB and $LUAJIT_INC (specified by the LUAJIT_LIB and LUAJIT_INC env)"
@@ -163,6 +178,7 @@ END
         fi
     fi
 fi
+fi # ngx_force_module
 
 if [ $ngx_found = yes ]; then
     CORE_INCS="$CORE_INCS $ngx_feature_path"
@@ -295,8 +311,8 @@ fi
 NGX_DTRACE_PROVIDERS="$NGX_DTRACE_PROVIDERS $ngx_addon_dir/dtrace/ngx_lua_provider.d"
 NGX_TAPSET_SRCS="$NGX_TAPSET_SRCS $ngx_addon_dir/tapset/ngx_lua.stp"
 
-USE_MD5=YES
-USE_SHA1=YES
+USE_MD5=NO
+USE_SHA1=NO
 
 CORE_INCS="$CORE_INCS $ngx_addon_dir/src/api"
 
diff -rupN lua-nginx-module.old/src/ngx_http_lua_socket_tcp.c lua-nginx-module.new/src/ngx_http_lua_socket_tcp.c
--- lua-nginx-module.old/src/ngx_http_lua_socket_tcp.c	2013-09-02 21:58:33.000000000 +0200
+++ lua-nginx-module.new/src/ngx_http_lua_socket_tcp.c	2015-06-12 14:39:57.283348675 +0200
@@ -3146,8 +3146,7 @@ ngx_http_lua_req_socket(lua_State *L)
     }
 
     lua_settop(L, 1);
-    lua_pushnil(L);
-    return 2;
+    return 1;
 }
 
 
