Index: prototype2/fpga/ahead/red_pitaya.sdk/SDK/SDK_Export/fsbl/src/main.c
===================================================================
--- prototype2/fpga/ahead/red_pitaya.sdk/SDK/SDK_Export/fsbl/src/main.c	(revision 811)
+++ prototype2/fpga/ahead/red_pitaya.sdk/SDK/SDK_Export/fsbl/src/main.c	(working copy)
@@ -104,6 +104,7 @@
 #include "xstatus.h"
 #include "fsbl_hooks.h"
 #include "xtime_l.h"
+#include "sleep.h"
 
 #ifdef XPAR_XWDTPS_0_BASEADDR
 #include "xwdtps.h"
@@ -1463,5 +1464,30 @@
 		return XST_FAILURE;
 	}
 
+	register u32* TestAddress;
+	register u32 TestValue = 0;
+
+	fsbl_printf(DEBUG_INFO, "Checking memory range: 0x%08x - 0x%08x\r\n",
+			DDR_START_ADDR, DDR_END_ADDR);
+	fsbl_printf(DEBUG_INFO, "Writing... ");
+
+	for (TestAddress = (u32*)DDR_START_ADDR; (u32)TestAddress < DDR_END_ADDR; ++TestAddress) {
+		*TestAddress = TestValue;
+		TestValue += 31;
+	}
+
+	fsbl_printf(DEBUG_INFO, "Reading... ");
+	TestValue = 0;
+
+	for (TestAddress = (u32*)DDR_START_ADDR; (u32)TestAddress < DDR_END_ADDR; ++TestAddress) {
+		while (*TestAddress != TestValue) {
+			fsbl_printf(DEBUG_INFO,"DDR error at *0x%08x: 0x%08x != 0x%08x\r\n",
+					TestAddress, *TestAddress, TestValue);
+			sleep(1);
+		}
+		TestValue += 31;
+	}
+	fsbl_printf(DEBUG_INFO,"PASSED!\r\n");
+
 	return XST_SUCCESS;
 }
