diff -rupN nginx-1.5.3.old/auto/lib/pcre/make nginx-1.5.3.new/auto/lib/pcre/make
--- nginx-1.5.3.old/auto/lib/pcre/make	2013-07-30 15:27:55.000000000 +0200
+++ nginx-1.5.3.new/auto/lib/pcre/make	2015-06-14 03:06:50.938241207 +0200
@@ -47,11 +47,12 @@ END
 
 $PCRE/pcre.h:	$PCRE/Makefile
 
+#JJJ - add LD and --host=arm for pcre configure
 $PCRE/Makefile:	$NGX_MAKEFILE
 	cd $PCRE \\
 	&& if [ -f Makefile ]; then \$(MAKE) distclean; fi \\
-	&& CC="\$(CC)" CFLAGS="$PCRE_OPT" \\
-	./configure --disable-shared $PCRE_CONF_OPT
+	&& CC="\$(CC)" CCLD="\$(LD)" CFLAGS="$PCRE_OPT" \\
+	./configure --disable-shared --host=arm $PCRE_CONF_OPT
 
 $PCRE/.libs/libpcre.a:	$PCRE/Makefile
 	cd $PCRE \\
diff -rupN nginx-1.5.3.old/src/core/ngx_log.c nginx-1.5.3.new/src/core/ngx_log.c
--- nginx-1.5.3.old/src/core/ngx_log.c	2013-07-30 15:27:55.000000000 +0200
+++ nginx-1.5.3.new/src/core/ngx_log.c	2015-06-14 03:06:50.946241116 +0200
@@ -343,9 +343,16 @@ ngx_log_init(u_char *prefix)
                                     NGX_FILE_DEFAULT_ACCESS);
 
     if (ngx_log_file.fd == NGX_INVALID_FILE) {
+
+        /* #3099: Be silent about the read only filesystem.
+         *        The error log is routed to /tmp (tmpfs) in nginx.conf.
+         */
+#if 0
         ngx_log_stderr(ngx_errno,
                        "[alert] could not open error log file: "
                        ngx_open_file_n " \"%s\" failed", name);
+#endif
+
 #if (NGX_WIN32)
         ngx_event_log(ngx_errno,
                        "could not open error log file: "
diff -rupN nginx-1.5.3.old/src/http/ngx_http_write_filter_module.c nginx-1.5.3.new/src/http/ngx_http_write_filter_module.c
--- nginx-1.5.3.old/src/http/ngx_http_write_filter_module.c	2013-07-30 15:27:55.000000000 +0200
+++ nginx-1.5.3.new/src/http/ngx_http_write_filter_module.c	2015-06-14 03:06:50.974240799 +0200
@@ -198,8 +198,8 @@ ngx_http_write_filter(ngx_http_request_t
             return NGX_OK;
         }
 
-        ngx_log_error(NGX_LOG_ALERT, c->log, 0,
-                      "the http output chain is empty");
+//        ngx_log_error(NGX_LOG_ALERT, c->log, 0,
+//                      "the http output chain is empty");
 
         ngx_debug_point();
 
diff -rupN nginx-1.5.3.old/src/os/unix/ngx_linux_config.h nginx-1.5.3.new/src/os/unix/ngx_linux_config.h
--- nginx-1.5.3.old/src/os/unix/ngx_linux_config.h	2013-07-30 15:27:55.000000000 +0200
+++ nginx-1.5.3.new/src/os/unix/ngx_linux_config.h	2015-06-14 03:06:50.982240709 +0200
@@ -13,6 +13,8 @@
 #define _GNU_SOURCE             /* pread(), pwrite(), gethostname() */
 #endif
 
+#define NGX_PTR_SIZE 4
+
 #define _FILE_OFFSET_BITS  64
 
 #include <sys/types.h>
