--- pl.dtsi	2016-05-03 13:31:51.983240381 +0200
+++ fpga2/sdk/dts/pl.dtsi	2016-05-03 13:27:53.503243061 +0200
@@ -18,12 +18,14 @@
 			interrupts = <0 30 4 0 29 4>;
 			reg = <0x80400000 0x10000>;
 			xlnx,include-sg ;
+			status = "disabled";
 			dma-channel@80400000 {
 				compatible = "xlnx,axi-dma-mm2s-channel";
 				dma-channels = <0x1>;
 				interrupts = <0 30 4>;
 				xlnx,datawidth = <0x20>;
 				xlnx,device-id = <0x0>;
+				status = "disabled";
 			};
 			dma-channel@80400030 {
 				compatible = "xlnx,axi-dma-s2mm-channel";
@@ -31,6 +33,7 @@
 				interrupts = <0 29 4>;
 				xlnx,datawidth = <0x20>;
 				xlnx,device-id = <0x0>;
+				status = "disabled";
 			};
 		};
 		axi_dma_1: dma@80410000 {
@@ -40,12 +43,14 @@
 			interrupts = <0 32 4 0 31 4>;
 			reg = <0x80410000 0x10000>;
 			xlnx,include-sg ;
+			status = "disabled";
 			dma-channel@80410000 {
 				compatible = "xlnx,axi-dma-mm2s-channel";
 				dma-channels = <0x1>;
 				interrupts = <0 32 4>;
 				xlnx,datawidth = <0x20>;
 				xlnx,device-id = <0x1>;
+				status = "disabled";
 			};
 			dma-channel@80410030 {
 				compatible = "xlnx,axi-dma-s2mm-channel";
@@ -53,6 +58,7 @@
 				interrupts = <0 31 4>;
 				xlnx,datawidth = <0x20>;
 				xlnx,device-id = <0x1>;
+				status = "disabled";
 			};
 		};
 		axi_dma_2: dma@80420000 {
@@ -62,12 +68,14 @@
 			interrupts = <0 34 4 0 33 4>;
 			reg = <0x80420000 0x10000>;
 			xlnx,include-sg ;
+			status = "disabled";
 			dma-channel@80420000 {
 				compatible = "xlnx,axi-dma-mm2s-channel";
 				dma-channels = <0x1>;
 				interrupts = <0 34 4>;
 				xlnx,datawidth = <0x10>;
 				xlnx,device-id = <0x2>;
+				status = "disabled";
 			};
 			dma-channel@80420030 {
 				compatible = "xlnx,axi-dma-s2mm-channel";
@@ -75,6 +83,7 @@
 				interrupts = <0 33 4>;
 				xlnx,datawidth = <0x10>;
 				xlnx,device-id = <0x2>;
+				status = "disabled";
 			};
 		};
 		axi_dma_3: dma@80430000 {
@@ -84,12 +93,14 @@
 			interrupts = <0 36 4 0 35 4>;
 			reg = <0x80430000 0x10000>;
 			xlnx,include-sg ;
+			status = "disabled";
 			dma-channel@80430000 {
 				compatible = "xlnx,axi-dma-mm2s-channel";
 				dma-channels = <0x1>;
 				interrupts = <0 36 4>;
 				xlnx,datawidth = <0x10>;
 				xlnx,device-id = <0x3>;
+				status = "disabled";
 			};
 			dma-channel@80430030 {
 				compatible = "xlnx,axi-dma-s2mm-channel";
@@ -97,12 +108,13 @@
 				interrupts = <0 35 4>;
 				xlnx,datawidth = <0x10>;
 				xlnx,device-id = <0x3>;
+				status = "disabled";
 			};
 		};
 		xadc_wiz_0: xadc_wiz@83c00000 {
 			compatible = "xlnx,xadc-wiz-3.3";
 			interrupt-parent = <&intc>;
 			interrupts = <0 52 4>;
 			reg = <0x83c00000 0x10000>;
 			xlnx,alarm-limit-r0 = <0xb5ed>;
 			xlnx,alarm-limit-r1 = <0x57e4>;
@@ -183,5 +196,78 @@
 			xlnx,vaux8 = <0x1>;
 			xlnx,vaux9 = <0x1>;
 		};
+		rprx_2: rprx@2 {
+		   compatible ="redpitaya,rprx";
+		   dmas = <&axi_dma_2 0
+			   &axi_dma_2 1>;
+		   dma-names = "axidma0", "axidma1";
+		   memory-region = <&rprx_reserverd>;
+			status = "disabled";
+		};
+		id: id@40000000 {
+			compatible = "generic-uio";
+			reg = <0x40000000 0x10000>;
+			status = "disabled";
+		};
+		muxctl: muxctl@40040000 {
+			compatible = "generic-uio";
+			reg = <0x40040000 0x10000>;
+			status = "disabled";
+		};
+		gpio: gpio@40080000 {
+			compatible = "generic-uio";
+			reg = <0x40080000 0x10000>;
+			status = "disabled";
+		};
+		led: led@400c0000 {
+			compatible = "generic-uio";
+			reg = <0x400c0000 0x10000>;
+			status = "disabled";
+		};
+		calib: calib@40100000 {
+			compatible = "generic-uio";
+			reg = <0x40100000 0x10000>;
+			status = "disabled";
+		};
+		pdm: pdm@40140000 {
+			compatible = "generic-uio";
+			reg = <0x40140000 0x10000>;
+			status = "disabled";
+		};
+		pwm: pwm@40180000 {
+			compatible = "generic-uio";
+			reg = <0x40180000 0x10000>;
+			status = "disabled";
+		};
+		asg: asg0@401c0000 {
+			compatible = "generic-uio";
+			reg = <0x401c0000 0x40000>;
+			status = "disabled";
+		};
+		asg1: asg1@40200000 {
+			compatible = "generic-uio";
+			reg = <0x40200000 0x40000>;
+			status = "disabled";
+		};
+		scope0: scope0@40240000 {
+			compatible = "generic-uio";
+			reg = <0x40240000 0x10000>;
+			status = "disabled";
+		};
+		scope1: scope1@40280000 {
+			compatible = "generic-uio";
+			reg = <0x40280000 0x10000>;
+			status = "disabled";
+		};
+		lg: lg@402c0000 {
+			compatible = "generic-uio";
+			reg = <0x402c0000 0x40000>;
+			status = "disabled";
+		};
+		la: la@40300000 {
+			compatible = "generic-uio";
+			reg = <0x40300000 0x10000>;
+			status = "disabled";
+		};
 	};
 };
