--- websocketpp-0.5.1/websocketpp/transport/asio/endpoint.hpp	2016-05-26 11:16:44.992474798 +0700
+++ websocketpp-0.5.1/websocketpp/transport/asio/endpoint2.hpp	2016-05-26 11:16:36.832474739 +0700
@@ -95,7 +95,7 @@ public:
     explicit endpoint()
       : m_io_service(NULL)
       , m_external_io_service(false)
-      , m_listen_backlog(boost::asio::socket_base::max_connections)
+      , m_listen_backlog(0)
       , m_reuse_addr(false)
       , m_state(UNINITIALIZED)
     {
