--- a/drivers/net/phy/intel-xway.c
+++ b/drivers/net/phy/intel-xway.c
@@ -243,6 +243,10 @@ static int xway_gphy_config_init(struct phy_device *phydev)
        /* Clear all pending interrupts */
        phy_read(phydev, XWAY_MDIO_ISTAT);

+        err = phy_write(phydev, 0x17, 0xb500);
+       if (err)
+               return err;
+
        /* Ensure that integrated led function is enabled for all leds */
        err = phy_write(phydev, XWAY_MDIO_LED,
                        XWAY_MDIO_LED_LED0_EN |
