From 34cf589c17cfda62a0ef1df6b2fd0fe7cb2fce45 Mon Sep 17 00:00:00 2001
From: Naveen Kumar Gaddipati <naveenku@xilinx.com>
Date: Tue, 25 Jul 2017 16:55:57 +0530
Subject: [PATCH] zynq:kernel:mempry reserve issue

Signed-off-by: Naveen Kumar Gaddipati <naveenku@xilinx.com>
---
 arch/arm/mach-zynq/common.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/mach-zynq/common.c b/arch/arm/mach-zynq/common.c
index 79ce23b..8177cdb 100644
--- a/arch/arm/mach-zynq/common.c
+++ b/arch/arm/mach-zynq/common.c
@@ -59,7 +59,7 @@ void __iomem *zynq_scu_base;
 static void __init zynq_memory_init(void)
 {
 	if (!__pa(PAGE_OFFSET))
-		memblock_reserve(__pa(PAGE_OFFSET), __pa(swapper_pg_dir));
+		memblock_reserve(__pa(PAGE_OFFSET), 0x80000);
 }
 
 static struct platform_device zynq_cpuidle_device = {
-- 
2.7.4

