
--- ./devicetree.dts
+++ ./devicetree.dts
@@ -154,9 +154,9 @@
 				#address-cells = <1>;
 				#size-cells = <0>;
 				phy0: phy@7 {
-					compatible = "marvell,88e1116r";
+					compatible = "lantiq,phy11g";
 					device_type = "ethernet-phy";
-					reg = <7>;
+					reg = <1>;
 				} ;
 			} ;
 		} ;
@@ -190,6 +190,16 @@
			xlnx,has-interrupt = <0x0>;
			xlnx,i2c-clk-freq-hz = <0x69f6bcb>;
			xlnx,i2c-reset = "";
+			#address-cells = <1>;
+			#size-cells = <0>;
+			eep@50 {
+				compatible = "24c64";
+				reg = <0x50>;
+			};
+			eep@51 {
+				compatible = "24c64";
+				reg = <0x51>;
+			};
 		} ;
		ps7_intc_dist_0: ps7-intc-dist@f8f01000 {
			compatible = "xlnx,ps7-intc-dist-1.00.a";
@@ -377,15 +387,15 @@
			reg = < 0xe0002000 0x1000 >;
			xlnx,usb-reset = "MIO 48";
		} ;
-		ps7_xadc: ps7-xadc@f8007100 {
-			compatible = "xlnx,ps7-xadc-1.00.a";
-			reg = < 0xf8007100 0x20 >;
-		} ;
-		ps7_xadc: ps7-xadc@f8007100 {
-			compatible = "xlnx,ps7-xadc-1.00.a";
-			interrupt-parent = <&ps7_scugic_0>;
-			interrupts = < 0 7 4 >;
-			reg = < 0xf8007100 0x20 >;
-		} ;
+//		ps7_xadc: ps7-xadc@f8007100 {
+//			compatible = "xlnx,ps7-xadc-1.00.a";
+//			reg = < 0xf8007100 0x20 >;
+//		} ;
+//		ps7_xadc: ps7-xadc@f8007100 {
+//			compatible = "xlnx,ps7-xadc-1.00.a";
+//			interrupt-parent = <&ps7_scugic_0>;
+//			interrupts = < 0 7 4 >;
+//			reg = < 0xf8007100 0x20 >;
+//		} ;
 	} ;
 } ;
