From fd385049cb686cdaba038d4f3420d6048ac51baf Mon Sep 17 00:00:00 2001
From: "crt.valentincic" <crt.valentincic@i-tech.com>
Date: Mon, 18 Nov 2013 14:05:43 +0100
Subject: [PATCH 01/11] Changed-serial-from-uart1-to-uart0 and phy addr=1

---
 include/configs/zynq_zed.h |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/include/configs/zynq_zed.h b/include/configs/zynq_zed.h
index 506b0ab..b11d956 100644
--- a/include/configs/zynq_zed.h
+++ b/include/configs/zynq_zed.h
@@ -20,9 +20,9 @@
 
 #define PHYS_SDRAM_1_SIZE (512 * 1024 * 1024)
 
-#define CONFIG_ZYNQ_SERIAL_UART1
+#define CONFIG_ZYNQ_SERIAL_UART0
 #define CONFIG_ZYNQ_GEM0
-#define CONFIG_ZYNQ_GEM_PHY_ADDR0	0
+#define CONFIG_ZYNQ_GEM_PHY_ADDR0	1
 
 #define CONFIG_SYS_NO_FLASH
 
-- 
1.7.9.5

