From 390bb667ed3cbeca830ac458fe0d7c6d6b73e359 Mon Sep 17 00:00:00 2001
From: "crt.valentincic" <crt.valentincic@redpitaya.com>
Date: Mon, 18 Nov 2013 14:08:14 +0100
Subject: [PATCH 02/11] Increased-PHY-autonegotiation-timeout-from-4-to-10s

---
 include/phy.h |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/phy.h b/include/phy.h
index 58ca273..86009e3 100644
--- a/include/phy.h
+++ b/include/phy.h
@@ -45,7 +45,7 @@
 #define PHY_10G_FEATURES	(PHY_GBIT_FEATURES | \
 				SUPPORTED_10000baseT_Full)
 
-#define PHY_ANEG_TIMEOUT	4000
+#define PHY_ANEG_TIMEOUT	10000
 
 
 typedef enum {
-- 
1.7.9.5

