--- u-boot-xlnx.orig/board/xilinx/zynq/board.c	2013-07-04 14:16:32.000000000 +0200
+++ u-boot-xlnx/board/xilinx/zynq/board.c	2013-12-13 01:57:58.566267405 +0100
@@ -30,7 +30,7 @@
 DECLARE_GLOBAL_DATA_PTR;
 
 /* Bootmode setting values */
-#define BOOT_MODES_MASK		0x0000000F
+#define BOOT_MODES_MASK		0x00000007 // only 3 bits have valid information
 #define QSPI_MODE		0x00000001
 #define NOR_FLASH_MODE		0x00000002
 #define NAND_FLASH_MODE		0x00000004
@@ -82,11 +82,11 @@
 	/* temporary hack to take USB out of reset til the is fixed
 	 * in Linux
 	 */
-	writel(0x80, 0xe000a204);
-	writel(0x80, 0xe000a208);
-	writel(0x80, 0xe000a040);
-	writel(0x00, 0xe000a040);
-	writel(0x80, 0xe000a040);
+	writel(0x10000, 0xe000a244);
+	writel(0x10000, 0xe000a248);
+	writel(0x10000, 0xe000a044);
+	writel(0x10000, 0xe000a044);
+	writel(0x10000, 0xe000a044);
 
 #ifdef CONFIG_FPGA
 	fpga_init();
