From d0ece111eea7d442b4e2cbfa9b80821e636ea2b4 Mon Sep 17 00:00:00 2001
From: "crt.valentincic" <crt.valentincic@redpitaya.com>
Date: Mon, 18 Nov 2013 14:58:31 +0100
Subject: [PATCH 07/11] Lantiq copyright changed

---
 drivers/net/phy/lantiq.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/phy/lantiq.c b/drivers/net/phy/lantiq.c
index bf49075..9d6955f 100644
--- a/drivers/net/phy/lantiq.c
+++ b/drivers/net/phy/lantiq.c
@@ -16,7 +16,7 @@
  * Foundation, Inc., 59 Temple Place, Suite 330, Boston,
  * MA 02111-1307 USA
  *
- * Copyright 2011 Freescale Semiconductor, Inc.
+ * Copyright 2013 Instrumentation Technologies d.d.
  * author Tomaž Beltram
  *
  */
-- 
1.7.9.5

