From 926acded3e8fbd5455c4d7afa3d02e670cb6f328 Mon Sep 17 00:00:00 2001
From: "crt.valentincic" <crt.valentincic@redpitaya.com>
Date: Mon, 18 Nov 2013 14:59:44 +0100
Subject: [PATCH 08/11] Fixed-10BASE-T-FULL-led-blink-speed-setting

---
 drivers/net/phy/lantiq.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/phy/lantiq.c b/drivers/net/phy/lantiq.c
index 9d6955f..4e9afc5 100644
--- a/drivers/net/phy/lantiq.c
+++ b/drivers/net/phy/lantiq.c
@@ -56,7 +56,7 @@ static int phy11g_config(struct phy_device *phydev)
 	phy_mmd_write(phydev, MMD_DEVAD_INT, INT_LED2H,
 		INT_LED_LINK1000 << 4 | INT_LED_LINK100);
 	phy_mmd_write(phydev, MMD_DEVAD_INT, INT_LED2L,
-		INT_LED_LINK10);
+		INT_LED_LINK10 << 4);
 
 	return 0;
 }
-- 
1.7.9.5

