From 9484eaf719c04ea8b7765476492c2edc8569060b Mon Sep 17 00:00:00 2001
From: "crt.valentincic" <crt.valentincic@redpitaya.com>
Date: Mon, 18 Nov 2013 15:20:22 +0100
Subject: [PATCH 09/11] Enabled-dhcp-command

---
 include/configs/zynq_common.h |    1 +
 1 file changed, 1 insertion(+)

diff --git a/include/configs/zynq_common.h b/include/configs/zynq_common.h
index 19a621f..0f6338b 100644
--- a/include/configs/zynq_common.h
+++ b/include/configs/zynq_common.h
@@ -96,6 +96,7 @@
 #ifdef CONFIG_SYS_ENET
 # define CONFIG_CMD_PING
 # define CONFIG_CMD_MII
+# define CONFIG_CMD_DHCP
 #else
 # undef CONFIG_CMD_NET
 # undef CONFIG_CMD_NFS
-- 
1.7.9.5

