From c3351eea488be630abca171107f511a36b332b92 Mon Sep 17 00:00:00 2001
From: "crt.valentincic" <crt.valentincic@redpitaya.com>
Date: Fri, 22 Nov 2013 14:23:27 +0100
Subject: [PATCH 5/5] Red-Pitaya-env-par-added-wp-eeprom-area-is-now-in-use

---
 include/configs/zynq_red_pitaya.h |   73 ++++++++++++-------------------------
 1 file changed, 24 insertions(+), 49 deletions(-)

diff --git a/include/configs/zynq_red_pitaya.h b/include/configs/zynq_red_pitaya.h
index 33c0868..43a2e22 100644
--- a/include/configs/zynq_red_pitaya.h
+++ b/include/configs/zynq_red_pitaya.h
@@ -211,9 +211,9 @@
 #  define CONFIG_ENV_EEPROM_IS_ON_I2C
 #endif
 
-#define CONFIG_ENV_OFFSET           0
-#define CONFIG_ENV_SIZE             2048 /* Total Size of Environment Sector */
-#define CONFIG_ENV_SECT_SIZE       CONFIG_ENV_SIZE
+#define CONFIG_ENV_OFFSET           (2048*3) /* WP area starts at last 1/4 of 8k eeprom */
+#define CONFIG_ENV_SIZE             1024 /* Total Size of Environment Sector */
+#define CONFIG_ENV_SECT_SIZE        CONFIG_ENV_SIZE
 #define CONFIG_CMD_SAVEENV         /* Command to save ENV to Flash */
 
 #endif
@@ -225,52 +225,27 @@
 #endif
 
 /* Default environment */
-#define CONFIG_EXTRA_ENV_SETTINGS   \
-    "ethaddr=00:26:33:14:50:00\0"   \
-    "kernel_image=uImage\0" \
-    "ramdisk_image=uramdisk.image.gz\0" \
-    "devicetree_image=devicetree.dtb\0" \
-    "bitstream_image=system.bit.bin\0"  \
-    "loadbit_addr=0x100000\0"   \
-    "kernel_size=0x500000\0"    \
-    "devicetree_size=0x20000\0" \
-    "ramdisk_size=0x5E0000\0"   \
-    "fdt_high=0x20000000\0" \
-    "initrd_high=0x20000000\0"  \
-    "mmc_loadbit_fat=echo Loading bitstream from SD/MMC/eMMC to RAM.. && " \
-        "mmcinfo && " \
-        "fatload mmc 0 ${loadbit_addr} ${bitstream_image} && " \
-        "fpga load 0 ${loadbit_addr} ${filesize}\0" \
-    "norboot=echo Copying Linux from NOR flash to RAM... && " \
-        "cp.b 0xE2100000 0x3000000 ${kernel_size} && " \
-        "cp.b 0xE2600000 0x2A00000 ${devicetree_size} && " \
-        "echo Copying ramdisk... && " \
-        "cp.b 0xE2620000 0x2000000 ${ramdisk_size} && " \
-        "bootm 0x3000000 0x2000000 0x2A00000\0" \
-    "qspiboot=echo Copying Linux from QSPI flash to RAM... && " \
-        "sf probe 0 0 0 && " \
-        "sf read 0x3000000 0x100000 ${kernel_size} && " \
-        "sf read 0x2A00000 0x600000 ${devicetree_size} && " \
-        "echo Copying ramdisk... && " \
-        "sf read 0x2000000 0x620000 ${ramdisk_size} && " \
-        "bootm 0x3000000 0x2000000 0x2A00000\0" \
-    "sdboot=echo Copying Linux from SD to RAM... && " \
-        "mmcinfo && " \
-        "fatload mmc 0 0x3000000 ${kernel_image} && " \
-        "fatload mmc 0 0x2A00000 ${devicetree_image} && " \
-        "fatload mmc 0 0x2000000 ${ramdisk_image} && " \
-        "bootm 0x3000000 0x2000000 0x2A00000\0" \
-    "nandboot=echo Copying Linux from NAND flash to RAM... && " \
-        "nand read 0x3000000 0x100000 ${kernel_size} && " \
-        "nand read 0x2A00000 0x600000 ${devicetree_size} && " \
-        "echo Copying ramdisk... && " \
-        "nand read 0x2000000 0x620000 ${ramdisk_size} && " \
-        "bootm 0x3000000 0x2000000 0x2A00000\0" \
-    "jtagboot=echo TFTPing Linux to RAM... && " \
-        "tftp 0x3000000 ${kernel_image} && " \
-        "tftp 0x2A00000 ${devicetree_image} && " \
-        "tftp 0x2000000 ${ramdisk_image} && " \
-        "bootm 0x3000000 0x2000000 0x2A00000\0"
+#define CONFIG_EXTRA_ENV_SETTINGS	\
+	"ethaddr=00:26:33:14:50:00\0"	\
+	"hw_rev=B\0"	\
+	"prod_date=12/22/13\0"	\
+	"serial=00000000\0"	\
+    "kernel_image=uImage\0"	\
+	"ramdisk_image=uramdisk.image.gz\0"	\
+	"devicetree_image=devicetree.dtb\0"	\
+	"bitstream_image=system.bit.bin\0"	\
+	"loadbit_addr=0x100000\0"	\
+	"kernel_size=0x500000\0"	\
+	"devicetree_size=0x20000\0"	\
+	"ramdisk_size=0x5E0000\0"	\
+	"fdt_high=0x20000000\0"	\
+	"initrd_high=0x20000000\0"	\
+	"sdboot=echo Copying Linux from SD to RAM... && " \
+		"mmcinfo && " \
+		"fatload mmc 0 0x3000000 ${kernel_image} && " \
+		"fatload mmc 0 0x2A00000 ${devicetree_image} && " \
+		"fatload mmc 0 0x2000000 ${ramdisk_image} && " \
+		"bootm 0x3000000 0x2000000 0x2A00000\0"
 
 /* default boot is according to the bootmode switch settings */
 #define CONFIG_BOOTCOMMAND      "run $modeboot"
-- 
1.7.9.5

