diff -Nru u-boot-xlnx.orig/tools/env/fw_env_main.c u-boot-xlnx/tools/env/fw_env_main.c
--- u-boot-xlnx.orig/tools/env/fw_env_main.c	2014-02-25 22:17:51.953917799 +0100
+++ u-boot-xlnx/tools/env/fw_env_main.c	2014-02-25 22:24:58.625918069 +0100
@@ -47,6 +47,7 @@
 #include <sys/file.h>
 #include <unistd.h>
 #include "fw_env.h"
+#include "version.h"
 
 #define	CMD_PRINTENV	"fw_printenv"
 #define CMD_SETENV	"fw_setenv"
@@ -61,7 +62,8 @@
 {
 
 	fprintf(stderr, "fw_printenv/fw_setenv, "
-		"a command line interface to U-Boot environment\n\n"
+		"a command line interface to U-Boot environment\n"
+		"version %s-%s\n\n"
 		"usage:\tfw_printenv [-n] [variable name]\n"
 		"\tfw_setenv [variable name] [variable value]\n"
 		"\tfw_setenv -s [ file ]\n"
@@ -80,7 +82,8 @@
 		"A variable without value will be dropped. It is possible\n"
 		"to put any number of spaces between the fields, but any\n"
 		"space inside the value is treated as part of the value "
-		"itself.\n\n"
+		"itself.\n\n",
+		VERSION_STR, REVISION_STR
 	);
 }
 
diff -Nru u-boot-xlnx.orig/tools/env/Makefile u-boot-xlnx/tools/env/Makefile
--- u-boot-xlnx.orig/tools/env/Makefile	2014-02-25 22:17:51.953917799 +0100
+++ u-boot-xlnx/tools/env/Makefile	2014-02-25 22:44:51.841917604 +0100
@@ -21,6 +21,10 @@
 # MA 02111-1307 USA
 #
 
+# Versioning system
+RP_VERSION ?= 0.00-0000
+RP_REVISION ?= devbuild
+
 include $(TOPDIR)/config.mk
 
 HOSTSRCS := $(SRCTREE)/lib/crc32.c  fw_env.c  fw_env_main.c
@@ -36,6 +40,8 @@
 		-DUSE_HOSTCC \
 		-DTEXT_BASE=$(TEXT_BASE)
 
+HOSTCPPFLAGS += -DRP_VERSION=$(RP_VERSION) -DRP_REVISION=$(RP_REVISION)
+
 ifeq ($(MTD_VERSION),old)
 HOSTCPPFLAGS += -DMTD_OLD
 endif
diff -Nru u-boot-xlnx.orig/tools/env/version.h u-boot-xlnx/tools/env/version.h
--- u-boot-xlnx.orig/tools/env/version.h	1970-01-01 01:00:00.000000000 +0100
+++ u-boot-xlnx/tools/env/version.h	2014-02-25 22:45:20.801917610 +0100
@@ -0,0 +1,30 @@
+/**
+ * $Id: version.h 1247 2014-02-22 19:41:08Z ales.bardorfer $
+ *
+ * @brief Red Pitaya simple version strings. To be embedded in binaries
+ *        at build time for SW traceability.
+ *
+ * @Author Ales Bardorfer <ales.bardorfer@redpitaya.com>
+ *
+ * (c) Red Pitaya  http://www.redpitaya.com
+ */
+
+#ifndef VERSION_H
+#define VERSION_H
+
+#define XSTR(s) STR(s)
+#define STR(s) #s
+ 
+#ifndef RP_VERSION
+#define VERSION_STR "0.00-0000"
+#else
+#define VERSION_STR XSTR(RP_VERSION)
+#endif
+ 
+#ifndef RP_REVISION
+#define REVISION_STR "unknown"
+#else
+#define REVISION_STR XSTR(RP_REVISION)
+#endif
+
+#endif /* VERSION_H */
