diff -Nru u-boot-xlnx.orig/arch/arm/cpu/armv7/zynq/ddrc.c u-boot-xlnx/arch/arm/cpu/armv7/zynq/ddrc.c
--- u-boot-xlnx.orig/arch/arm/cpu/armv7/zynq/ddrc.c	2014-03-25 23:56:38.064266305 +0100
+++ u-boot-xlnx/arch/arm/cpu/armv7/zynq/ddrc.c	2014-03-26 00:01:52.788266155 +0100
@@ -60,6 +60,7 @@
 		puts("Memory: ECC disabled\n");
 	}
 
-	if (width == ZYNQ_DDRC_CTRLREG_BUSWIDTH_16BIT)
+	if ((ecctype == ZYNQ_DDRC_ECC_SCRUBREG_ECCMODE_SECDED) &&
+	    (width == ZYNQ_DDRC_CTRLREG_BUSWIDTH_16BIT))
		gd->ram_size /= 2;
 }
