--- include/configs/zynq_red_pitaya.h	2017-01-17 20:59:19.000000000 +0700
+++ include/configs/zynq_red_pitaya.h	2018-08-31 15:32:51.000000000 +0700
@@ -23,8 +23,34 @@
 #define CONFIG_SYS_EEPROM_PAGE_WRITE_BITS	5
 #define CONFIG_SYS_EEPROM_SIZE			8192 /* Bytes */
 
+#undef CONFIG_ENV_SIZE
+#undef CONFIG_EXTRA_ENV_SETTINGS
+
 #define CONFIG_ENV_IS_IN_EEPROM
 #define CONFIG_ENV_SIZE		1024 /* Total Size of Environment Sector */
 #define CONFIG_ENV_OFFSET	(2048*3) /* WP area starts at last 1/4 of 8k eeprom */
 
+#define CONFIG_EXTRA_ENV_SETTINGS \
+    "bootcmd=run $modeboot\0" \
+    "bootdelay=3\0" \
+    "baudrate=115200\0" \
+    "ipaddr=10.10.70.102\0" \
+    "serverip=10.10.70.101\0" \
+    "prod_date=12/22/13\0" \
+    "kernel_image=uImage\0" \
+    "ramdisk_image=uramdisk.image.gz\0" \
+    "devicetree_image=devicetree.dtb\0" \
+    "bitstream_image=system.bit.bin\0" \
+    "loadbit_addr=0x100000\0" \
+    "kernel_size=0x500000\0" \
+    "devicetree_size=0x20000\0" \
+    "ramdisk_size=0x5E0000\0" \
+    "fdt_high=0x20000000\0" \
+    "initrd_high=0x20000000\0" \
+    "sdboot=echo Copying Linux from SD to RAM... && mmcinfo && fatload mmc 0 0x3000000 ${kernel_image} && fatload mmc 0 0x2A00000 ${devicetree_image} && fatload mmc 0 0x2000000 ${ramdisk_image} && bootm 0x3000000 0x2000000 0x2A00000\0" \
+    "ethaddr=88:88:88:88:88:88\0" \
+    "nav_code=0\0" \
+    "hw_rev=0\0" \
+    "serial=0\0"
+
 #endif /* __CONFIG_ZYNQ_RED_PITAYA_H */
