--- devicetree.dts	2015-03-26 13:37:21.256500338 +0100
+++ devicetree.dts	2015-03-26 13:59:33.923108675 +0100
@@ -154,9 +154,9 @@
 				#address-cells = <1>;
 				#size-cells = <0>;
 				phy0: phy@7 {
-					compatible = "marvell,88e1116r";
+					compatible = "lantiq,phy11g";
 					device_type = "ethernet-phy";
-					reg = <7>;
+					reg = <1>;
 				} ;
 			} ;
 		} ;
@@ -190,6 +190,12 @@
 			xlnx,has-interrupt = <0x0>;
 			xlnx,i2c-clk-freq-hz = <0x69f6bcb>;
 			xlnx,i2c-reset = "";
+			#address-cells = <1>;
+			#size-cells = <0>;
+			eep@50 {
+				compatible = "24c64";
+				reg = <0x50>;
+			};
 		} ;
 		ps7_intc_dist_0: ps7-intc-dist@f8f01000 {
 			compatible = "xlnx,ps7-intc-dist-1.00.a";
@@ -326,6 +332,11 @@
 			xlnx,has-ss1 = <0x0>;
 			xlnx,has-ss2 = <0x0>;
 			xlnx,spi-clk-freq-hz = <0x9ef21b0>;
+			spidev@0 {
+				compatible="spidev";
+				reg =<0>; //chipselect 0
+				spi-max-frequency= <50000000>;
+			} ;
 		} ;
 		ps7_ttc_0: ps7-ttc@f8001000 {
 			compatible = "xlnx,ps7-ttc-1.00.a", "cdns,ttc";
@@ -364,15 +375,15 @@
 			reg = < 0xe0002000 0x1000 >;
 			xlnx,usb-reset = "MIO 48";
 		} ;
-		ps7_xadc: ps7-xadc@f8007100 {
-			compatible = "xlnx,ps7-xadc-1.00.a";
-			reg = < 0xf8007100 0x20 >;
-		} ;
-		ps7_xadc: ps7-xadc@f8007100 {
-			compatible = "xlnx,ps7-xadc-1.00.a";
-			interrupt-parent = <&ps7_scugic_0>;
-			interrupts = < 0 7 4 >;
-			reg = < 0xf8007100 0x20 >;
-		} ;
+//		ps7_xadc: ps7-xadc@f8007100 {
+//			compatible = "xlnx,ps7-xadc-1.00.a";
+//			reg = < 0xf8007100 0x20 >;
+//		} ;
+//		ps7_xadc: ps7-xadc@f8007100 {
+//			compatible = "xlnx,ps7-xadc-1.00.a";
+//			interrupt-parent = <&ps7_scugic_0>;
+//			interrupts = < 0 7 4 >;
+//			reg = < 0xf8007100 0x20 >;
+//		} ;
 	} ;
 } ;
