--- release1/fpga/vivado/red_pitaya.sdk/SDK/SDK_Export/fsbl/src/main.c	2014-03-26 00:29:10.000000000 +0100
+++ release1/fpga/vivado/red_pitaya.sdk/SDK/SDK_Export/fsbl/src/main.c	2014-03-26 01:42:11.788265551 +0100
@@ -127,6 +127,7 @@
 #include "xstatus.h"
 #include "fsbl_hooks.h"
 #include "xtime_l.h"
+#include "sleep.h"
 
 #ifdef XPAR_XWDTPS_0_BASEADDR
 #include "xwdtps.h"
@@ -1512,5 +1513,30 @@
 		return XST_FAILURE;
 	}
 
+	register u32* TestAddress;
+	register u32 TestValue = 0;
+
+	fsbl_printf(DEBUG_INFO, "Checking memory range: 0x%08x - 0x%08x\r\n",
+			DDR_START_ADDR, DDR_END_ADDR);
+	fsbl_printf(DEBUG_INFO, "Writing... ");
+
+	for (TestAddress = (u32*)DDR_START_ADDR; (u32)TestAddress < DDR_END_ADDR; ++TestAddress) {
+		*TestAddress = TestValue;
+		TestValue += 31;
+	}
+
+	fsbl_printf(DEBUG_INFO, "Reading... ");
+	TestValue = 0;
+
+	for (TestAddress = (u32*)DDR_START_ADDR; (u32)TestAddress < DDR_END_ADDR; ++TestAddress) {
+		while (*TestAddress != TestValue) {
+			fsbl_printf(DEBUG_INFO,"DDR error at *0x%08x: 0x%08x != 0x%08x\r\n",
+					TestAddress, *TestAddress, TestValue);
+			sleep(1);
+		}
+		TestValue += 31;
+	}
+	fsbl_printf(DEBUG_INFO,"PASSED!\r\n");
+
 	return XST_SUCCESS;
 }
--- release1/fpga/vivado/red_pitaya.sdk/SDK/SDK_Export/fsbl/Debug/src/subdir.mk	2014-03-26 00:29:10.000000000 +0100
+++ release1/fpga/vivado/red_pitaya.sdk/SDK/SDK_Export/fsbl/Debug/src/subdir.mk	2014-03-26 03:07:51.840264927 +0100
@@ -66,14 +66,14 @@
 src/%.o: ../src/%.c
 	@echo 'Building file: $<'
 	@echo 'Invoking: ARM gcc compiler'
-	arm-xilinx-eabi-gcc -Wall -O0 -g3 -c -fmessage-length=0 -I../../fsbl_bsp/ps7_cortexa9_0/include -MMD -MP -MF"$(@:%.o=%.d)" -MT"$(@:%.o=%.d)" -o "$@" "$<"
+	arm-xilinx-eabi-gcc -Wall -O0 -g3 -c -fmessage-length=0 -I../../fsbl_bsp/ps7_cortexa9_0/include -MMD -MP -MF"$(@:%.o=%.d)" -MT"$(@:%.o=%.d)" -DFSBL_DEBUG_INFO -o "$@" "$<"
 	@echo 'Finished building: $<'
 	@echo ' '
 
 src/%.o: ../src/%.S
 	@echo 'Building file: $<'
 	@echo 'Invoking: ARM gcc compiler'
-	arm-xilinx-eabi-gcc -Wall -O0 -g3 -c -fmessage-length=0 -I../../fsbl_bsp/ps7_cortexa9_0/include -MMD -MP -MF"$(@:%.o=%.d)" -MT"$(@:%.o=%.d)" -o "$@" "$<"
+	arm-xilinx-eabi-gcc -Wall -O0 -g3 -c -fmessage-length=0 -I../../fsbl_bsp/ps7_cortexa9_0/include -MMD -MP -MF"$(@:%.o=%.d)" -MT"$(@:%.o=%.d)" -DFSBL_DEBUG_INFO -o "$@" "$<"
 	@echo 'Finished building: $<'
 	@echo ' '
 
