--- connman-1.19.orig/src/service.c	2013-10-14 04:30:03.000000000 +0200
+++ connman-1.19/src/service.c	2014-03-03 10:12:00.615290522 +0100
@@ -5454,6 +5454,14 @@
 
 	default_changed();
 
+        /* Fork a script when link becomes ready */
+        const size_t cmd_size = 256;
+        char cmd[cmd_size];
+        snprintf(cmd, cmd_size, "/usr/lib/connman/scripts/connman-default.sh %s &",
+                 state2string(service->state));
+        DBG("Calling %s", cmd);
+        system(cmd);
+        
 	return 0;
 }
 
