diff -Nur linux-xlnx-orig/drivers/net/phy/phy.c linux-xlnx/drivers/net/phy/phy.c
--- linux-xlnx-orig/drivers/net/phy/phy.c	2013-07-01 14:42:18.000000000 +0200
+++ linux-xlnx/drivers/net/phy/phy.c	2013-12-23 11:36:38.916609717 +0100
@@ -854,6 +854,11 @@
 				phydev->state = PHY_RUNNING;
 				netif_carrier_on(phydev->attached_dev);
 				phydev->adjust_link(phydev->attached_dev);
+			} else if (0 == phydev->link_timeout--) {
+				/* If we have the magic_aneg bit,
+				 * we try again */
+				if (phydev->drv->flags & PHY_HAS_MAGICANEG)
+					needs_aneg = 1;
 			}
 			break;
 		case PHY_FORCING:
